components/component.go

```diff
@@ -80,13 +83,52 @@ type ManifestsConfig struct {
 }
 
 type ComponentInterface interface {
-	ReconcileComponent(ctx context.Context, cli client.Client, resConf *rest.Config, owner metav1.Object, DSCISpec *dsciv1.DSCInitializationSpec, currentComponentStatus bool) error
+	ReconcileComponent(ctx context.Context, cli client.Client, resConf *rest.Config, logger logr.Logger,
+		owner metav1.Object, DSCISpec *dsciv1.DSCInitializationSpec, currentComponentStatus bool) error
 	Cleanup(cli client.Client, DSCISpec *dsciv1.DSCInitializationSpec) error
 	GetComponentName() string
 	GetManagementState() operatorv1.ManagementState
 	SetImageParamsMap(imageMap map[string]string) map[string]string
 	OverrideManifests(platform string) error
 	UpdatePrometheusConfig(cli client.Client, enable bool, component string) error
+	ConfigLogger(logger logr.Logger, component string, dscispec *dsciv1.DSCInitializationSpec) logr.Logger
+}
+
+// by default, ConfigLogger with jsonEncoder, logLevel=Info, stackTraceLevel=Error
+// when set in DSCI: devel Mode defaults(encoder=consoleEncoder,logLevel=Debug,stackTraceLevel=Info with lower case for key
+// when set in DSCI: prod  Mode defaults(encoder=jsonEncoder,   logLevel=Info, stackTraceLevel=Error with upper case for key.
+func (c *Component) ConfigLogger(logger logr.Logger, component string, dscispec *dsciv1.DSCInitializationSpec) logr.Logger {
```

Comment:
Could we unify how the loggers are created between here and main.go?

Reply:
did some updates, not sure if this is what you meant for "unify" ?

Reply:
unify = reuse code / call same function. Make a change in 1 location. Ensure they behave the same etc.. :)

Reply:
in this case, it cannot be a method for DSC, gonna make it is util function